### ISSUE TITLE
`configure` tries to assert that no configuration exists in config folder

### ISSUE THREAD

**Issue by DropD** [greschd/aiida_pytest#5]
It seems the part of the code that backs up the configuration fails, if any configuration already exists. This is probably not intended, since it results in backing up only inexistent configurations.

```python
@contextmanager
    def reset_after_run():
    ...
    assert not os.path.isfile(os.path.join(config_folder, 'config.json'))
```

The following error is generated from the contributed aiida-vasp tests:

```
_________________________________________________________________________________________________________________________ ERROR at setup of test_vasp __________________________________________________________________________________________________________________________

pytestconfig = <_pytest.config.Config object at 0x10a7a2250>
config_dict = {'codes': [{'default_plugin': 'vasp.vasp', 'description': 'VASP', 'label': 'vasp', 'prepend_text': 'source /users/gres...: '/home/greschd/software/VASP-5.4.1/potpaw_PBE.54', 'group_description': 'VASP 5.4 potpaw_PBE', 'group_name': 'pbe'}]}

    @pytest.fixture(scope='session')
    def configure(pytestconfig, config_dict):
        config = copy.deepcopy(config_dict)
        with temporary.temp_dir() as td, PGTest(max_connections=100) as pgt:
>           with reset_after_run():

../../aiida_pytest/aiida_pytest/configure.py:52:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/Cellar/python/2.7.13_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/contextlib.py:17: in __enter__
    return self.gen.next()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

    @contextmanager
    def reset_after_run():
        config_folder = os.path.expanduser(aiida.common.setup.AIIDA_CONFIG_FOLDER)
        config_save_folder = os.path.join(
            os.path.dirname(config_folder), '.aiida~'
        )
        reset_config(config_folder, config_save_folder)
>       assert not os.path.isfile(os.path.join(config_folder, 'config.json'))
E       AssertionError

../../aiida_pytest/aiida_pytest/configure.py:101: AssertionError
```

**Comment by greschd** [comment]
This was in fact intended, at least for the initial version of the code. The reasoning is that I want to have an additional layer keeping me from messing up production environments. So the goal is to have the tests produce the full config.

Maybe this can be changed now -- but I'm not sure backup/restore is airtight. In particular, it could be that restore fails when the test process is killed. Another concern is what happens if production and test are run simultaneously on the same config folder.

**Comment by greschd** [comment]
Fixed in #11.
